flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/LocationPreferenceSlotSelectionStrategy.java

```diff
@@ -122,7 +121,7 @@ protected abstract Optional<SlotInfoAndLocality> selectWithoutLocationPreference
             @Nonnull ResourceProfile resourceProfile);
 
     protected abstract double calculateCandidateScore(
-            int localWeigh, int hostLocalWeigh, double taskExecutorUtilization);
+            int localWeigh, int hostLocalWeigh, SlotInfoAndResources candidate);
```

Comment:
How about change this method to
```
protected abstract double calculateCandidateScore(
            int localWeigh, int hostLocalWeigh, Supplier<Double> taskExecutorUtilizationSupplier);
```
As it stands, passing `SlotInfoAndResources` seems redundant.